extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -0,0 +1,336 @@
+      return false;
+    }
+    const actions = Services.actionServiceForDoc(element);
+    return !actions.hasAction(element, 'tap');
+  }
+}
+
+
+/**
+ * @param {number} renderWidth
+ * @param {number} renderHeight
+ * @param {number} naturalWidth
+ * @param {number} naturalHeight
+ * @param {number} vw
+ * @param {number} vh
+ * @return {boolean}
+ * @visibleForTesting
+ */
+export function meetsSizingCriteria(
+  renderWidth,
+  renderHeight,
+  naturalWidth,
+  naturalHeight,
+  vw,
+  vh) {
+
+  const viewportArea = vw * vh;
+  const naturalArea = naturalWidth * naturalHeight;
+  const renderArea = renderWidth * renderHeight;
+
+  const isShrunk =
+    (naturalArea / renderArea) >= RENDER_AREA_RATIO;
+
+  const isCoveringSignificantArea =
+    (renderArea / viewportArea) >= VIEWPORT_AREA_RATIO;
+
+  const isLargerThanViewport = naturalWidth > vw || naturalHeight > vh;
+
+  return isShrunk || isLargerThanViewport || isCoveringSignificantArea;
+}
+
+
+/** @visibleForTesting */
+export class Scanner {
+
+  /**
+   * @param {!Document} doc
+   * @return {<!Array<!Promise<!Element>>}
+   */
+  static getAllImages(doc) {
+    const imgs = toArray(doc.querySelectorAll('amp-img')).filter(img =>
+      !img.hasAttribute(LIGHTBOXABLE_ATTR) &&
+      !img.hasAttribute(VISITED_ATTR));
+
+    imgs.forEach(img => {
+      img.setAttribute(VISITED_ATTR, '');
+    });
+
+    return imgs.map(whenLoadedOrNull);
+  }
+}
+
+
+/**
+ * @param {!Element} element
+ * @return {boolean}
+ * @visibleForTesting
+ */
+export function meetsCriteria(element) {
+  return Criteria.meetsAll(element);
+}
+
+
+/**
+ * @param {!IArrayLike<T>} haystack
+ * @param {function(T):boolean} needleCb
+ * @return {?T}
+ * @template T
+ */
+function find(haystack, needleCb) {
+  for (let i = 0; i < haystack.length; i++) {
+    if (needleCb(haystack[i])) {
+      return haystack[i];
+    }
+  }
+  return null;
+}
+
+/** @visibleForTesting */
+export class Schema {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @return {?string}
+   */
+  static getDocumentType(ampdoc) {
+    const schemaTags = ampdoc.getHeadNode().querySelectorAll(
+        'script[type="application/ld+json"]');
+
+    if (schemaTags.length <= 0) {
+      return null;
+    }
+
+    const {canonicalUrl} = Services.documentInfoForDoc(ampdoc);
+
+    for (let i = 0; i < schemaTags.length; i++) {
+      const schemaTag = schemaTags[i];
+      const parsed = tryParseJson(schemaTag./*OK*/innerText);
+
+      if (parsed &&
+        'url' in parsed &&
+        parsed['url'] == canonicalUrl) {
+
+        return parsed['@type'];
+      }
+    }
+
+    return null;
+  }
+}
+
+
+/**
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {boolean}
+ */
+function usesLightboxExplicitly(ampdoc) {
+  const scriptTags = ampdoc.getHeadNode().querySelectorAll('script');
+
+  const currentScriptTag = find(scriptTags,
+      t => t.getAttribute('custom-element') == REQUIRED_EXTENSION);
```

Comment:
This should be handled by `src/service/extensions-impl.js`

Reply:
Yeah, not sure why it's not provided as a public API. I don't know what the best way to expose this may be, since there is [some extension `holder` logic I'm not particularly familiar with](https://github.com/ampproject/amphtml/blob/master/src/service/extensions-impl.js#L603).